java/spi.java.hints/src/org/netbeans/spi/java/hints/JavaFixUtilities.java

```diff
@@ -285,9 +285,11 @@ private static String defaultFixDisplayName(CompilationInfo info, Map<String, Tr
         return "Rewrite to " + replaceTarget.replace("♦", "$").replace("<", "&lt;");
     }
 
-    // could this be leaf.toString()?
     private static String treeToString(TreePath tp, CompilationInfo info) {
         Tree leaf = tp.getLeaf();
+        if (leaf.getKind() == Kind.VARIABLE) {
+            return ((VariableTree) leaf).getName().toString();
+        }
```

Comment:
```
$mods$ $type $name = new $T();
```
lets say we have the tp for `$name` the old code would have essentially called tp.getLeaf().toString() which would be the whole statement. This will only return the identifier (`$name`).

I am not sure if there are other trees which need similar treatment.